sql/colexec: TestRouterOutputAddBatch failed

[(sql/colexec).TestRouterOutputAddBatch failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1904948&tab=buildLog) on [master@c3bb2bce3d69d0ceef20797d856a6ac107a3ef47](https://github.com/cockroachdb/cockroach/commits/c3bb2bce3d69d0ceef20797d856a6ac107a3ef47):

```
=== RUN   TestRouterOutputAddBatch
--- FAIL: TestRouterOutputAddBatch (1.80s)
=== RUN   TestRouterOutputAddBatch/QuarterSelection/memoryLimit=0_B
    --- FAIL: TestRouterOutputAddBatch/QuarterSelection/memoryLimit=0_B (0.00s)
        routers_test.go:171: 
            	Error Trace:	routers_test.go:171
            	Error:      	Not equal: 
            	            	expected: true
            	            	actual  : false
            	Test:       	TestRouterOutputAddBatch/QuarterSelection/memoryLimit=0_B

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestRouterOutputAddBatch PKG=./pkg/sql/colexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRouterOutputAddBatch.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/colexec/routers_test.go
+++ b/pkg/sql/colexec/routers_test.go
@@ -140,6 +140,10 @@ func TestRouterOutputAddBatch(t *testing.T) {
 	defer cleanup()
 
 	for _, tc := range testCases {
+		if len(tc.selection) == 0 {
+			// No data to work with, probably due to a low coldata.BatchSize.
+			continue
+		}
 		for _, mtc := range memoryTestCases {
 			t.Run(fmt.Sprintf("%s/memoryLimit=%s", tc.name, humanizeutil.IBytes(mtc.bytes)), func(t *testing.T) {
 				// Clear the testAllocator for use.

